builtin/logical/mysql/path_role_create.go

```diff
@@ -51,9 +52,15 @@ func (b *backend) pathRoleCreateRead(
 	}
 
 	// Generate our username and password. MySQL limits user to 16 characters
-	displayName := req.DisplayName
-	if len(displayName) > 10 {
-		displayName = displayName[:10]
+	displayName := name
+	ul, ok := data.GetOk("username_length")
+	if ok == true {
+		usernameLength = ul.(int)
+	} else {
+		usernameLength = 10
+	}
+	if len(displayName) > usernameLength {
```

Comment:
Are we not supposed to refer to the value in the `role` here, for usernameLength?